src/payment/InvoiceProcessor.ts

```diff
@@ -0,0 +1,286 @@
+import { Allocation, Invoice } from "ya-ts-client/dist/ya-payment";
+import { Yagna } from "../utils";
+import { YagnaApi, YagnaOptions } from "../utils/yagna/yagna";
+import { Decimal, Numeric } from "decimal.js-light";
+
+export type InvoiceAcceptResult =
+  | {
+      invoiceId: string;
+      allocation: Allocation;
+      success: true;
+      amount: string;
+      dryRun: boolean;
+    }
+  | {
+      invoiceId: string;
+      allocation: Allocation;
+      success: false;
+      amount: string;
+      reason: unknown;
```

Comment:
Both variants of the type differ in the `reason` field this could be unified to a single type with `reason?:unknown`

Reply:
`reason` is only defined if `success: false` so IMO a discriminated union like this one works best here